portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,19 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+async function showServerTime() {
+  const responseFromServer = await fetch('/date');
+  const textFromResponse = await responseFromServer.text();
+
+  const dateContainer = document.getElementById('date-container');
+  dateContainer.innerText = textFromResponse;
+}
+
+async function generateRandomResponse() {
+  const responseFromServer = await fetch('/random');
+  const textFromResponse = await responseFromServer.text();
+
+  const dateContainer = document.getElementById('random-generator');
```

Comment:
the variable name here should be something other than dataContainer, something along line of "random.."